parser/src/test/java/com/microsoft/gctoolkit/parser/ConcurrentMarkSweepPhaseParserRulesTest.java

```diff
@@ -30,13 +32,21 @@ public void testCMSParseRules() {
 
     /* Code that is useful when testing individual records */
 
-    private final boolean debugging = Boolean.getBoolean("microsoft.debug");
+    private static final boolean DEBUGGING;
```

Comment:
We seem to be duplicating this code block in a lot of places..

Reply:
True. I replaced all that with a GCToolkit.LOG_DEBUG_MESSAGE(Supplier<String>) method. Using supplier means there is no eval of the code that builds the message unless debugging is enabled.